### ISSUE TITLE
problem

### ISSUE THREAD

**Issue by eain3314** [Neerajj9/CSRNet-keras#14]
密度图的ground-truth是根据标签生成的，然后根据密度图积分求和得到人数，对吗？
但是如果用一个随便找的图片进行测试的时候，没有标签，那怎么生成的密度图？
能解答一下吗

**Comment by aditya-malte** [comment]
Hello,
As I see it, the Google translation of the above is :
"The ground-truth of the density map is generated based on the label, and then the sum is obtained according to the density map integral, right?
But if you use a random picture to test, there is no label, then how to generate a density map?
Can you answer it?"

Your first question talks about the training phase when annotated/labeled training data is available, and your understanding regarding the same is correct. The CNN is trained to map images to density map.

However, the second phase talks about the inference phase when the model itself is supposed to generate the density maps. The trained CNN then generates the density map after passing the image through itself. Thus, no labeled data is required in this phase
Thank you

**Comment by eain3314** [comment]
谢谢  我还有一个问题 ，就是测试的时候，根据测试产生的密度图如何估算出来的人数？

**Comment by aditya-malte** [comment]
Hello,
As per Google Translate:
" Thanks, I still have a question. How is the number of people estimated
based on the density map generated during the test? "

Coming back to your question, the answer is simply taking the summation of
the density map so generated.
Basically, the density maps is nothing but a 2d matrix where values near 1
mean a high crowd density and values near 0 mean otherwise.
We simply take the summation of all the elements of this matrix to obtain
the count.
Thank you.

**Comment by eain3314** [comment]
能详细的解释一下如何训练和测试数据的流程吗？

**Comment by aditya-malte** [comment]
Hello,
Please read the documentation provided to understand the same.

**Comment by eain3314** [comment]
训练的时候是训练的人头位置吗？还是别的东西？

**Comment by aditya-malte** [comment]
I honestly did not understand what you said(and nor did Google). Please resend it in my native language-English.
Thank you
